core/internal/runmetric/runmetric.go

```diff
@@ -115,21 +115,23 @@ func (mh *MetricHandler) UpdateMetrics(
 	return mh.createGlobMetrics(history)
 }
 
-// InsertStepMetrics inserts an automatic step metric for every defined
-// metric with step_sync set to true.
+// InsertStepMetrics inserts an automatic step metric for every metric
+// that has a step metric defined and is not already present in the
+// history.
```

Comment:
the original comment should still be correct. The new version goes into too much detail about the implementation.